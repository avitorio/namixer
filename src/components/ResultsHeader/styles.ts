import styled, { css } from 'styled-components'

export const ResultsHeader = styled.div`
  ${({ theme }) => css`
    display: flex;
    background-color: ${theme.colors.mainBg};
    padding: ${theme.spacings.xsmall} 0;
    margin: ${theme.spacings.xsmall} 0;
    width: 100%;
    max-width: 96rem;
    align-items: center;
    justify-content: space-between;
    position: sticky;
    top: -1px;
    z-index: 2;
  `}
`
