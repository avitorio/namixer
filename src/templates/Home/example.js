export const domainList = [
  { domain: 'google.com', isAvailable: true },
  { domain: 'goozgle.com', isAvailable: true },
  { domain: 'goozglez.com', isAvailable: true },
  { domain: 'gzoogle.com', isAvailable: true },
  { domain: 'gozogle.com', isAvailable: true }
]
