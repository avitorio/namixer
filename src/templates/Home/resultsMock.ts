import { SearchResults } from '.'

export const results = <SearchResults[]>[
  {
    domain: 'BrazilTv.com',
    status: 'taken'
  },
  {
    domain: 'BrazilCo.com',
    status: 'taken'
  },
  {
    domain: 'BrazilMe.com',
    status: 'taken'
  },
  {
    domain: 'BrazilUp.com',
    status: 'taken'
  },
  {
    domain: 'BrazilFx.com',
    status: 'taken'
  },
  {
    domain: 'BrazilHq.com',
    status: 'taken'
  },
  {
    domain: 'BrazilLy.com',
    status: 'taken'
  },
  {
    domain: 'BrazilDb.com',
    status: 'available'
  },
  {
    domain: 'BrazilEx.com',
    status: 'taken'
  },
  {
    domain: 'BrazilIo.com',
    status: 'taken'
  },
  {
    domain: 'BrazilDr.com',
    status: 'available'
  },
  {
    domain: 'BrazilHd.com',
    status: 'taken'
  },
  {
    domain: 'BrazilWeb.com',
    status: 'taken'
  }
]
