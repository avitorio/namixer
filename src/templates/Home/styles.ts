import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  color: #fff;
`

export const MainSearch = styled.section`
  ${({ theme }) => css`
    margin: 0 calc(-${theme.grid.gutter} / 2) ${theme.spacings.large};

    ${media.greaterThan('medium')`
      max-width: 96rem;
      width: 100%;
      display: flex;
      align-items: center;
      justify-content: space-between;
    `}
  `}
`
