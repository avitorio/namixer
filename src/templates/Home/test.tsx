import 'match-media-mock'

jest.mock('templates/Base', () => ({
  __esModule: true,
  default: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Mock Base">{children}</div>
  }
}))

describe('<Home />', () => {
  it('should render banner and showcases', () => {
    // render(<Home />)
  })
})
