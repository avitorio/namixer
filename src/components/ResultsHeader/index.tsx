import Toggle from 'components/Toggle'
import { useEffect, useState } from 'react'
import { SearchResults } from 'templates/Home'
import * as S from './styles'

export type ResultsHeaderProps = {
  results: SearchResults[]
  setHideTaken: (value: boolean | ((prevVar: boolean) => boolean)) => void
}

const ResultsHeader = ({ results = [], setHideTaken }: ResultsHeaderProps) => {
  const [found, setFound] = useState(0)
  const [taken, setTaken] = useState(0)
  console.log(results)
  useEffect(() => {
    const found = results.filter(({ status }) => status === 'available').length
    const taken = results.filter(({ status }) => status === 'taken').length
    setFound(found)
    setTaken(taken)
  }, [results])

  return (
    <>
      {results.length > 0 && (
        <S.ResultsHeader>
          <Toggle
            onCheck={setHideTaken}
            label="Only show available domains."
            labelFor="hideTaken"
          />
          <span>{`${found} domains available. ${taken} taken.`}</span>
        </S.ResultsHeader>
      )}
    </>
  )
}

export default ResultsHeader
