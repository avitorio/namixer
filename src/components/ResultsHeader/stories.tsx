import { Story, Meta } from '@storybook/react/types-6-0'
import ResultsHeader, { ResultsHeaderProps } from '.'
import { results } from 'templates/Home/resultsMock'

export default {
  title: 'ResultsHeader',
  component: ResultsHeader,
  args: {
    results,
    setHideTaken: () => true
  }
} as Meta

export const Default: Story<ResultsHeaderProps> = (args) => (
  <ResultsHeader {...args} />
)
