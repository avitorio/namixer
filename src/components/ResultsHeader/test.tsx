import { render } from 'utils/test-utils'

import { results } from 'templates/Home/resultsMock'
import ResultsHeader from '.'

describe('<ResultsHeader />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <ResultsHeader
        results={results}
        setHideTaken={() => false}
        hideTaken={false}
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
