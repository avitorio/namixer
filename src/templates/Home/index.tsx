import io from 'socket.io-client'
import { useEffect, useState } from 'react'
import Base from 'templates/Base'
import { Container } from 'components/Container'
import SearchField from 'components/SearchField'
import Button from 'components/Button'
import { socket } from '../../config/web-sockets'

import { domainList } from './example'

import * as S from './styles'

const Home = () => {
  const [values, setValues] = useState({ word: '' })

  const handleInput = (field: string, value: string) => {
    setValues((s) => ({ ...s, [field]: value }))
  }
  const onClick = () => {
    console.log('yo')
    socket.emit('what', values, (error) => {
      console.log(values.word)
    })
  }
  return (
    <Base>
      <Container>
        <S.Wrapper>
          {socket ? (
            <div className="chat-container">
              <h1>Connected</h1>
            </div>
          ) : (
            <div>Not Connected</div>
          )}
          <S.MainSearch>
            <SearchField
              name="word"
              placeholder="Type in a word"
              type="text"
              onInput={(v) => handleInput('word', v)}
            />
            <Button size="xlarge" onClick={onClick}>
              Search Domains
            </Button>
          </S.MainSearch>
          <ul>
            {domainList.map(
              (result: { domain: string; isAvailable: boolean }) => (
                <li key={result.domain}>
                  <a href="https://google.com">{result.domain}</a>
                </li>
              )
            )}
          </ul>
        </S.Wrapper>
      </Container>
    </Base>
  )
}

export default Home
