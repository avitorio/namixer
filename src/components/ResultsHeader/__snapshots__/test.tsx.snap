// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsHeader /> should render the heading 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  position: relative;
  gap: 1ch;
}

.c1 span {
  --offset: 0.15em;
  --diameter: 1.2em;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: calc(var(--diameter) * 2 + var(--offset) * 2);
  height: calc(var(--diameter) + var(--offset) * 2);
  box-sizing: content-box;
  border: 0.1em solid rgb(0 0 0 / 0.2);
  position: relative;
  border-radius: 100vw;
  background-color: #E65A6D;
  -webkit-transition: 250ms;
  transition: 250ms;
  cursor: pointer;
}

.c1 span::before {
  content: '';
  width: var(--diameter);
  height: var(--diameter);
  border-radius: 50%;
  box-sizing: border-box;
  border: 0.1 solid rgb(0 0 0 / 0.2);
  position: absolute;
  z-index: 2;
  top: 50%;
  left: var(--offset);
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  background-color: #fff;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1 input {
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.c1 input:focus + span {
  outline: 1px dotted #212121;
  outline: 1px auto -webkit-focus-ring-color;
}

.c1 input:focus:focus:not(:focus-visible) + span {
  outline: 0;
}

.c1 input:checked + span {
  background-color: #3cd3c1;
}

.c1 input:checked + span::before {
  -webkit-transform: translate(100%,-50%);
  -ms-transform: translate(100%,-50%);
  transform: translate(100%,-50%);
}

.c1 input:disabled + span {
  opacity: 0.6;
  -webkit-filter: grayscale(40%);
  filter: grayscale(40%);
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #030517;
  padding: 1.6rem 0;
  margin: 1.6rem 0;
  width: 100%;
  max-width: 96rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: -webkit-sticky;
  position: sticky;
  top: -1px;
  z-index: 2;
}

@media (prefers-reduced-motion:reduce) {
  .c1 span {
    -webkit-transition-duration: 0ms;
    transition-duration: 0ms;
  }
}

<div
  class="c0"
>
  <label
    class="c1"
    for="toggle"
  >
    <input
      id="hideTaken"
      name="toggle"
      type="checkbox"
      value=""
    />
    <span
      hidden=""
    >
      <svg
        aria-hidden="true"
        class="Toggle__icon Toggle__icon--checkmark"
        fill="none"
        focusable="false"
        height="14"
        viewBox="0 0 18 14"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.08471 10.6237L2.29164 6.83059L1 8.11313L6.08471 13.1978L17 2.28255L15.7175 1L6.08471 10.6237Z"
          fill="currentcolor"
          stroke="currentcolor"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="Toggle__icon Toggle__icon--cross"
        fill="none"
        focusable="false"
        height="13"
        viewBox="0 0 13 13"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.167 0L6.5 4.667L1.833 0L0 1.833L4.667 6.5L0 11.167L1.833 13L6.5 8.333L11.167 13L13 11.167L8.333 6.5L13 1.833L11.167 0Z"
          fill="currentcolor"
        />
      </svg>
    </span>
    Only show available domains.
  </label>
  <span>
    2 domains available. 11 taken.
  </span>
</div>
`
